src/databricks/labs/ucx/hive_metastore/pipelines_migrate.py

```diff
@@ -0,0 +1,160 @@
+        src_pipeline_id: str,
+        target_catalog_name: str | None = None,
+        target_schema_name: str | None = None,
+        target_pipeline_name: str | None = None,
+    ) -> "PipelineRule":
+        return cls(
+            workspace_name=workspace_name,
+            src_pipeline_id=src_pipeline_id,
+            target_catalog_name=target_catalog_name,
+            target_schema_name=target_schema_name,
+            target_pipeline_name=target_pipeline_name,
+        )
+
+    @classmethod
+    def initial(cls, workspace_name: str, catalog_name: str, pipeline: PipelineInfo) -> "PipelineRule":
+        return cls(
+            workspace_name=workspace_name,
+            target_catalog_name=catalog_name,
+            src_pipeline_id=pipeline.pipeline_id,
+            target_pipeline_name=pipeline.pipeline_name,
+            target_schema_name=None,
+        )
+
+
+@dataclass
+class PipelineToMigrate:
+    src: PipelineInfo
+    rule: PipelineRule
+
+    def __hash__(self):
+        return hash(self.src)
+
+    def __eq__(self, other):
+        return isinstance(other, PipelineToMigrate) and self.src == other.src
+
+
+class PipelineMapping:
+    FILENAME = "pipeline_mapping.csv"
+
+    def __init__(
+        self,
+        installation: Installation,
+        ws: WorkspaceClient,
+        sql_backend: SqlBackend,
+    ):
+        self._installation = installation
+        self._ws = ws
+        self._sql_backend = sql_backend
+
+    @staticmethod
+    def current_pipelines(pipelines: PipelinesCrawler, workspace_name: str, catalog_name: str) -> Generator:
+        pipeline_snapshot = list(pipelines.snapshot(force_refresh=True))
+        if not pipeline_snapshot:
+            msg = "No pipelines found."
+            raise ValueError(msg)
+        for pipeline in pipeline_snapshot:
+            yield PipelineRule.initial(workspace_name, catalog_name, pipeline)
+
+    def load(self) -> list[PipelineRule]:
+        try:
+            return self._installation.load(list[PipelineRule], filename=self.FILENAME)
+        except NotFound:
+            msg = "Please create pipeline mapping file"
+            raise ValueError(msg) from None
+
+    def get_pipelines_to_migrate(self, _pc) -> list[PipelineToMigrate]:
+        rules = self.load()
+        # Getting all the source tables from the rules
+        pipelines = list(_pc.snapshot())
+        pipelines_to_migrate = []
+
+        for rule in rules:
+            for pipeline in pipelines:
+                if pipeline.pipeline_id == rule.src_pipeline_id:
+                    pipelines_to_migrate.append(PipelineToMigrate(pipeline, rule))
+
+        return pipelines_to_migrate
+
+
+class PipelinesMigrator:
+    def __init__(self, ws: WorkspaceClient, pipelines_crawler: PipelinesCrawler, pipeline_mapping: PipelineMapping):
+        self._ws = ws
+        self._pipeline_crawler = pipelines_crawler
+        self._pipeline_mapping = pipeline_mapping
+
+    def migrate_pipelines(self) -> None:
```

Comment:
This method looks redundant

Reply:
I will either rename this or the private method.

Reply:
OKe keep this one :)